src/client.cpp

```diff
@@ -71,6 +71,7 @@ void Client::receiveTask() {
             publishEvent(ClientEvent::DISCONNECTED, disconnectionMessage);
             return;
         } else {
+            receivedMessage[numOfBytesReceived] = '\0';
```

Comment:
Thank you for the PR. This might solve the issue, but I think it should be done differently. The `numOfBytesReceived` should be published along with the `receivedMessage` on the next line of code. This provides greater flexibility for the end-user.